src/main/java/org/jenkinsci/plugins/saml/SamlEncryptionData.java

```diff
@@ -46,11 +47,11 @@ public String getKeystorePath() {
     }
 
     public String getKeystorePassword() {
-        return keystorePassword;
+        return Secret.toString(keystorePassword);
```

Comment:
I think this is a configuration breakage. Old passwords were stored as plain text so this decryption will break them.